scripts/diagnostics.sh

```diff
@@ -8,138 +8,187 @@
-    fi
+function print_failure_and_exit()
+{
+    print_environment
+    printf "\nThere was an error while $(print_red "$@")\033[0m.
 
-    # Static
-    which mpicc &> /dev/null
-    if [ "$?" -eq 0 ]; then
-        write_to_stdout "\nMPICC:\nwhich mpicc:\n\t`which mpicc`\nmpicc -show:\n\t`mpicc -show`\n\nCOMPILER `echo $(mpicc -show | cut -d\  -f1)`:\n`$(mpicc -show | cut -d\  -f1) --version`"
-    else
-        write_to_stdout "\nMPICC WRAPPER NOT FOUND"
-    fi
+Please report the entirety of the output of diagnostics on
+this terminal to either YOUR existing post OR a new post on
+https://github.com/idaholab/moose/discussions\n"
+    exit 1
 }
 
-diagnose_modules() {
-    type module &> /dev/null
-    if [ "$?" -eq 0 ]; then
-        write_to_stdout "\nMODULES:\n`module list 2>&1`"
-    else
-        write_to_stdout "\nMODULES NOT AVAILABLE"
-    fi
+function print_sep()
+{
+    printf '\n'
+    printf '#%.0s' {1..75}
+    printf '\n'
 }
 
-diagnose_root() {
-    if [ $(id -u) = 0 ]; then write_to_stdout "RUNNING AS ROOT"; fi
+function print_red()
+{
+    printf "\033[1;31m$@\033[0m"
 }
 
-diagnose_environment() {
-    write_to_stdout "\nENVIRONMENT:\n`env | sort`"
+function print_orange()
+{
+    printf "\033[38;5;202m$@\033[0m"
 }
 
-diagnose_cpus() {
-    if [ `uname` == "Darwin" ]; then
-        CPUS=`/usr/sbin/sysctl -n hw.ncpu`
-    else
-        CPUS=`cat /proc/cpuinfo | grep processor | wc -l`
-    fi
-    write_to_stdout "\nCPU Count: `echo $CPUS`"
+function print_green()
+{
+    printf "\033[1;32m$@\033[0m"
 }
 
-diagnose_memory() {
+function create_tmp()
+{
+    if [ -n "$CTMP_DIR" ]; then echo $CTMP_DIR; return; fi
     if [ `uname` == "Darwin" ]; then
-        PAGE_SIZE=`vm_stat | grep "page size of" | sed 's/[^0-9]*//g'`
-        MEMORY_FREE=`vm_stat | grep free | awk -v page_size=$PAGE_SIZE '{ print (page_size * $3) / 1048576}'`
+        local TMP_DIR=${TMP_DIR:-`mktemp -d -t MOOSE`}
     else
-        MEMORY_FREE=`awk '/MemFree/ { printf "%.3f \n", $2/1024 }' /proc/meminfo`
+        local TMP_DIR=${TMP_DIR:-`mktemp -d -t MOOSE.XXXXXXXX`}
     fi
-    write_to_stdout "\nMemory Free: `echo $MEMORY_FREE` MB"
+    echo $TMP_DIR
 }
 
-diagnose_arch() {
-    if [ `uname` == "Darwin" ]; then
-        SYSTEM=`uname -v`
-    else
-        which lsb_release &> /dev/null
-        if [ "$?" -eq 0 ]; then
-            SYSTEM="`lsb_release -a`"
-        else
-            if [ -f /etc/system-release ]; then
-                SYSTEM="`cat /etc/system-release`"
-            else
-                SYSTEM="Not Available"
-            fi
-        fi
-    fi
-    write_to_stdout "\nSystem Arch: `echo $SYSTEM`"
+function print_help()
+{
+    args=("-h|--help"\
+          "-f|--full-build"\
+          "-r|--run-checks"\
+          "-a|--active-environment"\
+          "-c|--conda-channel")
+
+    args_about=("Print this message and exit."\
+                "Build everything: PETSc, libMesh, WASP, MOOSE."\
+                "Run Checks."\
```

Comment:
Here is what is printed for `args_about`. This is a bash array and it is how you represent each argument's about string.

```pre
❯ ./diagnostics.sh -h

Syntax:
        ./diagnostics.sh

Options:

   -h|--help                 Print this message and exit.
   -f|--full-build           Build everything: PETSc, libMesh, WASP, MOOSE.
   -r|--run-checks           Run Checks.
   -a|--active-environment   Do not use a pristine Conda environment, use current environment instead.
   -c|--conda-channel        Prioritize MOOSE packages with this channel (default is public).

Influencial Environment Variables:

        METHODS
        METHOD
        MOOSE_JOBS
        REQUESTS_CA_BUNDLE
        SSL_CERT_FILE
        CURL_CA_BUNDLE

Supplying no arguments prints useful environment information.
```

I think you are after supplying no arguments? The default method? Looks like I misspelled Influential... I'll fix that!

Reply:
```pre
--run-checks
```

What do you want to call the two small tests we run (these guys):

```pre
function test_always_ok()
{
    ./run_tests --${METHOD} -i always_ok -x  | tee fail.log &>/dev/null
    if [ "$?" != "0" ]; then
        if [ -f tests/test_harness/test_harness.always_ok.FAIL.txt ]; then
            cat tests/test_harness/test_harness.always_ok.FAIL.txt
        else
            cat fail.log
        fi
        print_failure_and_exit 'running always_ok'
    fi
    print_green 'OK: Serial Tests\n'
}

function test_parallel()
{
    ./run_tests --${METHOD} -i always_ok -p 2 &>/dev/null
    if [ "$?" != "0" ]; then
        cat tests/test_harness/test_harness.always_ok.FAIL.txt
        print_failure_and_exit 'running parallel test'
    fi
    print_green 'OK: Parallel Tests\n'
}
```

These two tests are all we need to make a lot of diagnostic assumptions. I am not running `run_tests`. I only do that on a full build. Might as well after a full build, since that takes a while.

Reply:
ah so my comment is just saying let's say which checks in that docstring
so either 'Run Diagnostics' or 'Run Diagnostics and Tests'
which options is running the tests? is it this one?

Reply:
How's this?

```pre
    -f|--full-build    Build PETSc, libMesh, WASP, MOOSE, and then run entire TestHarness suite.
    -r|--run-checks    Build MOOSE and then run aggregated tests.
```

What shall we call these? `-r|--run-checks` will not catch INL's network issues. But it can (or rather TODO: will) detect most issues found on our https://mooseframework.inl.gov/help/troubleshooting.html page. That is my intent anyway.

As for `-f|--full-build`, this option finds issues with a machines network such as ours when subjected to VPN, etc. As well as issues that would have been discovered on our troubleshooting page. I am concerned if I should not run the entire `./run_tests` suite though. I figured "why not", since it takes quite a while to build everything anyway. 🤷‍♂️

Reply:
yup those work for me
maybe " test suite." instead of TestHarness (name of the python tool)